packages/issuer-rest/lib/IssuerTokenEndpoint.ts

```diff
@@ -37,16 +40,51 @@ export const handleTokenRequest = <T extends object>({
       })
     }
 
+    if (request.headers.authorization && request.headers.authorization.startsWith('DPoP ') && !request.headers.DPoP) {
+      return sendErrorResponse(response, 400, {
+        error: TokenErrorResponse.invalid_request,
+        error_description: 'DPoP header is required',
+      })
+    }
+
+    let dPoPJwk: JWK | undefined
+    if (request.headers.dpop) {
+      if (!dPoPVerifyJwtCallback) {
+        return sendErrorResponse(response, 400, {
+          error: TokenErrorResponse.invalid_request,
+          error_description: 'DPOP is not supported',
+        })
+      }
+
+      try {
+        const fullUrl = request.protocol + '://' + request.get('host') + request.originalUrl
```

Comment:
Host is not always defined or the same url. Host can be 0.0.0.0 while you use a public url. I think this needs to be configured somewhere (i've bad issues with this in credo when deploying to a https server url)

Reply:
Hmmm, I am not sure about it. I don't mind adding it as a parameter to the handleTokenRequest, but it looks weird to have it pass as a parameter. 
> @nklomp You also had a question about this. 
> Please doublecheck whether request.protocol isn't already including the trailing :
> https://github.com/expressjs/express/issues/4697#issue-994058715
> https://stackoverflow.com/questions/10183291/how-to-get-the-full-url-in-express/10185427#10185427

For now, I changed the implementation so that the access token endpoint URL can be passed; if passed as an argument, we use it. Otherwise, we fall back to the above behavior. We can also enforce passing it. What do you think?